prime-router/src/test/kotlin/azure/SubmissionFunctionTests.kt

```diff
@@ -131,24 +143,39 @@ class SubmissionFunctionTests : Logging {
         )
     )
 
+    val dataProvider = MockDataProvider { emptyArray<MockResult>() }
+    val connection = MockConnection(dataProvider)
+    val accessSpy = spyk(DatabaseAccess(connection))
+
+    private fun makeEngine(metadata: Metadata, settings: SettingsProvider): WorkflowEngine {
+        return spyk(
+            WorkflowEngine.Builder().metadata(metadata).settingsProvider(settings).databaseAccess(accessSpy).build()
+        )
+    }
+
+    @BeforeEach
+    fun reset() {
```

Comment:
Mock setups "leak" across tests, which can lead to very weird and confusing test errors.